src/classes/CON_Preferred_TDTM.cls

```diff
@@ -48,15 +48,17 @@ public class CON_Preferred_TDTM extends TDTM_Runnable {
 
             // While this logic is not performing any DML currently, the recursion flag is utilized here to be ready for any future logic changes.
             TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_Preferred_TDTM, true);
+            
+            List<Trigger_Handler__c> preferredPhoneOnOff = [SELECT Active__c from Trigger_Handler__c where Class__c = 'CON_PreferredPhone_TDTM'];
 
             for(SObject so : newlist) {
                 Contact contact = (Contact)so;
 
-                // Process the prefered Email
+                // Process the Prefered Email
                 CON_Email.processPreferredEmail(contact, oldlist);
                 
-                // Process the Preferred Phone
-                if(!String.isBlank(contact.PreferredPhone__c)) {
+                // Process the Preferred Phone.
+                if(!String.isBlank(contact.PreferredPhone__c) && preferredPhoneOnOff != NULL && preferredPhoneOnOff.size() > 0 && preferredPhoneOnOff[0].Active__c == False) {
```

Comment:
Hmmmmmm - red flag here - what if the customer deleted the TDTM class? If they've deleted the class, I'd expected we'd actually want this to run, but with the way the code is written it would not.

Maybe something like `if(!String.isBlank(contact.PreferredPhone__c) && (preferredPhoneOnOff == null || preferedPhoneOnOff.size() == 0 || preferredPhoneOnOff[0].Active__c == False)) {`

Reply:
^ This should run this code if the other TDTM class is disabled or if it doesn't exist.

Reply:
@skosgi1 See my updated comment above - this is still a red flag in my opinion. We want this code to run if 1) the customer doesn't have the CON_PreferredPhone_TDTM class or 2) if the CON_PreferredPhone_TDTM class is disabled. Right now only #2 is handled.

Reply:
@skosgi1  Just kidding - ignore this. I was looking at outdated code. You've already fixed this - resolving thread.